rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,535 @@
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.exchanges.data_structures import AssetMovement, MarginPosition, Trade
+from rotkehlchen.exchanges.exchange import ExchangeInterface, ExchangeQueryBalances
+from rotkehlchen.exchanges.utils import pair_symbol_to_base_quote
+from rotkehlchen.history.deserialization import deserialize_price
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_fee,
+    deserialize_fval,
+)
+from rotkehlchen.types import (
+    ApiKey,
+    ApiSecret,
+    AssetMovementCategory,
+    ExchangeAuthCredentials,
+    Fee,
+    Location,
+    Timestamp,
+    TimestampMS,
+    TradeType,
+)
+from rotkehlchen.utils.misc import ts_ms_to_sec, ts_now, ts_now_in_ms, ts_sec_to_ms
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.history.events.structures.base import HistoryEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+PAGINATION_LIMIT: Final = 50
+# RECEIV_WINDOW specifies how long an HTTP request is valid.
+# It is also used to prevent replay attacks. its unit in ms
+# https://bybit-exchange.github.io/docs/v5/guide#parameters-for-authenticated-endpoints
+RECEIV_WINDOW: Final = '5000'
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def bybit_symbol_to_base_quote(symbol: str) -> tuple[AssetWithOracles, AssetWithOracles]:
+    """Turns a bybit symbol product into a base/quote asset tuple
+
+    - Can raise UnprocessableTradePair if symbol is in unexpected format
+    - Case raise UnknownAsset if any of the pair assets are not known to rotki
+    """
+    five_letter_assets = ('SUSHI', '1INCH', 'MATIC', 'TENET', 'LOOKS', 'SFUND')
+    try:
+        base_asset, quote_asset = pair_symbol_to_base_quote(
+            symbol=symbol,
+            asset_deserialize_fn=asset_from_bybit,
+            five_letter_assets=five_letter_assets,
+        )
+    except UnprocessableTradePair:
+        # bybit has special pairs with perpetuals/shorts of the tokens
+        if '2L' in symbol:
+            splitted_symbol = symbol.split('2L')
+        elif '2S' in symbol:
+            splitted_symbol = symbol.split('2S')
+        elif '2' in symbol:
+            splitted_symbol = symbol.split('2')
+        else:
+            raise
+
+        if len(splitted_symbol) == 2:
+            base_asset = asset_from_bybit(splitted_symbol[0])
+            quote_asset = asset_from_bybit(splitted_symbol[1])
+        else:
+            raise
+
+    return base_asset, quote_asset
+
+
+class Bybit(ExchangeInterface):
+    def __init__(
+            self,
+            name: str,
+            api_key: ApiKey,
+            secret: ApiSecret,
+            database: 'DBHandler',
+            msg_aggregator: 'MessagesAggregator',
+    ):
```

Comment:
you mentioned you would add a docstring explaining the unified account and other concepts. I don't see it.